src/PooledArrays.jl

```diff
@@ -60,12 +70,17 @@ const PooledMatrix{T,R} = PooledArray{T,R,2}
 ##############################################################################
 
 # Echo inner constructor as an outer constructor
-function PooledArray(refs::RefArray{R}, invpool::Dict{T,R}, pool=_invert(invpool)) where {T,R}
-    PooledArray{T,eltype(R),ndims(R),R}(refs, invpool, pool)
+PooledArray(refs::RefArray{R}, invpool::Dict{T,R}, pool::Vector{T}, cow::Bool,
+            lock::Ref{Threads.ReentrantLock}) where {T,R} =
+    PooledArray{T,eltype(R),ndims(R),R}(refs, invpool, pool, cow, lock)
```

Comment:
Maybe do this to simplify the rest of the code?
```suggestion
PooledArray(refs::RefArray{R}, invpool::Dict{T,R}, pool::Vector{T}, cow::Bool,
            lock::Threads.ReentrantLock) where {T,R} =
    PooledArray{T,eltype(R),ndims(R),R}(refs, invpool, pool, cow, Ref(lock))
```

It could also make sense to use defaults for the new arguments. That could avoid breaking external code that may rely on these.